File: conformance/bundles/09-expected-message-mismatch/boom.steps.go
// Go sibling of boom.steps.ts (bundle 09-expected-message-mismatch).
package fixture

import (
	"errors"

	"github.com/varar-dev/varar-go/varar"
)

func Register(s *varar.Steps) {
	// Fails with a message that does NOT contain the expected substring
	// "expected message", so the expected-failure is NOT satisfied → the
	// example fails.
	s.Stimulus("I always boom", func(state varar.Value, args []varar.Value) (*varar.Value, error) {
		return nil, errors.New("actual different error")
	})
}

func State() varar.Value {
	return varar.NullValue
}

File: conformance/bundles/09-expected-message-mismatch/boom.steps.cs
// C# sibling of boom.steps.ts / .rs (bundle 09-expected-message-mismatch).
using Varar;
using Varar.Core;

namespace Varar.Corpus.B09;

public static class BoomSteps
{
    public static void Register(Steps s)
    {
        // Throws a message that does NOT contain the expected substring "expected message", so the
        // expected-failure is NOT satisfied → the example fails.
        s.Stimulus("I always boom", state => throw new HandlerException("actual different error"));
    }

    public static Value State() => Value.Null;
}
